packages/client/lib/sync/execution/vmexecution.ts

```diff
@@ -112,10 +117,26 @@ export class VMExecution extends Execution {
             // set as new head block
             headBlock = block
           } catch (error) {
+            // TODO: determine if there is a way to differentiate between the cases
+            // a) a bad block is served by a bad peer -> delete the block and restart sync
+            //    sync from parent block
+            // b) there is a consensus error in the VM -> stop execution until the
+            //    consensus error is fixed
+            //
+            // For now only option b) is implemented, atm this is a very likely case
+            // and the implemented behavior helps on debugging.
+            // Option a) would likely need some block comparison of the same blocks
+            // received by different peer to decide on bad blocks
+            // (minimal solution: receive block from 3 peers and take block if there is
+            // is equally served from at least 2 peers)
```

Comment:
I'm not sure if this is the way to go, if we have malicious attackers they can just setup multiple nodes to force us to execute bad blocks.

Reply:
See https://github.com/ethereumjs/ethereumjs-monorepo/pull/1075/files#r567690721 for an answer as well.